### ISSUE TITLE
UTF8 encoding issue when loading a saved *.tikz file

### ISSUE THREAD

**Issue by 0xACE** [tikzit/tikzit#75]
Hello, I don't have time to review this problem any further tonight. Anyhow I'm on version 2.1.4 and when I save my file and later open it in `tikzit` it will load the non-ASCII characters incorrectly and then it will fail to generate a preview.

**Minimal sample file:**
```
\begin{tikzpicture}
        \begin{pgfonlayer}{nodelayer}
                \node [style=none] (0) at (-1.75, 2) {Läkare};
        \end{pgfonlayer}
\end{tikzpicture}
```

**Error:**
```
! Package inputenc Error: Unicode character � (U+FFFD)
(inputenc)                not set up for use with LaTeX.

See the inputenc package documentation for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.11 		\node [style=none] (0) at (-1.75, 2) {L�
                                                 kare};
!  ==> Fatal error occurred, no output PDF file produced!
Transcript written on preview.log.




pdflatex RETURNED AN ERROR
```

**Comment by akissinger** [comment]
Thanks for pointing this out. The parser uses flex, which doesn't support UTF-8. It looks like switching to something like RE/flex could fix the problem, but I hesitate to do that because it doesn't seem to have official packages e.g. for Ubuntu. This could make life difficult for package managers and people wanting to install from source.

**Comment by 0xACE** [comment]
Yeah I was heading to bed as I wrote that so I couldn't delve any deeper. Judging by your response I realize your concern with this conflict.

Therefore I propose alleviation for this problem:

What do you think about implementing a primitive `normalizing` function that turns some characters into some latex code that respects the ASCII limitations? I realize this can be a difficult problem, but at least we could alleviate some issues.

I found this link: https://tug.org/pipermail/texhax/2010-January/014117.html

Which handles the swedish characters: `ÅÄÖåäö`:
```
å  => \aa
Å => \AA
ä  => \"a
Ä => \"A
ö  => \"o
Ö => \"O
```

Meaning that before we save the file, we can change all occurrences of those character to a ASCII friendly version, ensure it loads fine on the next session.

We could probably find other common characters who could use the same treatment.

Again this is not a solution, but it alleviates the pain when using `tikzit` in non-english languages.

What do you think about this idea?

**Comment by gspr** [comment]
> Thanks for pointing this out. The parser uses flex, which doesn't support UTF-8. It looks like switching to something like RE/flex could fix the problem, but I hesitate to do that because it doesn't seem to have official packages e.g. for Ubuntu. This could make life difficult for package managers and people wanting to install from source.

Indeed, with my Debian tikzit maintainer hat on, it would be a major problem. Are there really no good parser generators out there with UTF-8 support that are widely available? (I'm completely ignorant about this).

**Comment by 0xACE** [comment]
![cap](https://user-images.githubusercontent.com/1482942/82743995-5ebcaa80-9d62-11ea-992b-1f1890690b3f.png)

I'm currently running with a decoding/encoding function that handles the letters that affect me.

It can easily be applied for french, german and spanish. Probably some other languages too, but this implementation is really just a simple hack...
